Avoid shutting down twice, especially because of a boost::thread join change in 1.52
https://svn.boost.org/trac/boost/ticket/7875

--- clients/cpp/roscpp/src/libros/poll_manager.cpp~	2012-07-17 00:18:36.000000000 +0200
+++ clients/cpp/roscpp/src/libros/poll_manager.cpp	2013-04-12 01:25:15.000000000 +0200
@@ -66,6 +66,8 @@
 
 void PollManager::shutdown()
 {
+  if (shutting_down_) return;
+
   shutting_down_ = true;
   if (thread_.get_id() != boost::this_thread::get_id())
   {
